includedBuild/gradleConfiguration/src/main/kotlin/com/badoo/reaktive/publish/PublishConfigurationPlugin.kt

```diff
@@ -3,37 +3,117 @@ package com.badoo.reaktive.publish
+        target.plugins.apply(MavenPublishPlugin::class.java)
         val taskConfigurationMap = createConfigurationMap(target)
-        createFilteredPublishToMavenLocalTask(target)
-        setupLocalPublishing(target, taskConfigurationMap)
-        setupBintrayPublishingInformation(target)
-        setupBintrayPublishing(target, taskConfigurationMap)
+        disablePublishingTasks(target, taskConfigurationMap)
+        createFilteredPublishTasks(target)
+        setupPublishing(target)
+        setupSign(target)
+        createEmptySourcesJar(target)
     }
 
-    private fun createFilteredPublishToMavenLocalTask(target: Project) {
-        // Create empty task with enabled publish tasks only
-        // We can't use regular publishToMavenLocal
-        target.tasks.register(TASK_FILTERED_PUBLISH_TO_MAVEN_LOCAL) {
-            dependsOn(project.tasks.matching { it is AbstractPublishToMaven && it.enabled })
+    private fun setupPublishing(project: Project) {
+        val publishing = project.extensions.getByType(PublishingExtension::class.java)
+        publishing.repositories {
+            maven {
+                name = "sonatype"
+                val repositoryId = project.findProperty("sonatype.repository")
+                url = URI.create(
+                    if (repositoryId != null) {
+                        "https://oss.sonatype.org/service/local/staging/deployByRepositoryId/$repositoryId/"
+                    } else {
+                        "https://oss.sonatype.org/service/local/staging/deploy/maven2/"
+                    }
+                )
+                credentials {
+                    username = project.findProperty("sonatype.username").toString()
+                    password = project.findProperty("sonatype.password").toString()
+                }
+            }
+        }
+        publishing.publications.withType(MavenPublication::class.java).configureEach {
+            pom {
+                name.set(project.name)
+                description.set(project.description)
+                url.set("https://github.com/badoo/Reaktive")
+                licenses {
+                    license {
+                        name.set("Apache-2.0")
+                        url.set("https://github.com/badoo/Reaktive/blob/master/LICENSE")
+                    }
+                }
+                scm {
+                    connection.set("scm:git:ssh://github.com/badoo/Reaktive.git")
+                    developerConnection.set("scm:git:ssh://github.com/badoo/Reaktive.git")
+                    url.set("https://github.com/badoo/Reaktive")
+                }
+            }
+        }
+    }
+
+    private fun setupSign(project: Project) {
+        val inMemoryKey = project.findProperty("signing.key")?.toString()
+        // Constants from PgpSignatoryFactory.PROPERTIES
+        val password = project.findProperty("signing.password")?.toString()
+        val keyFile = project.findProperty("signing.secretKeyRingFile")?.toString()
```

Comment:
Isn't in-memory key is enough? Why do we need the file as well?

Reply:
In case if we want to publish from personal computer. It is easier just to specify file we have rather than passing the key through command line.